assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -108,9 +108,7 @@ const SavedPaymentMethodOptions = ( { onSelect } ) => {
 	const updateToken = useCallback(
 		( token ) => {
 			if ( token !== '0' ) {
-				// @todo this will need updated when the signature changes to
-				// allow no billing data included.
-				setPaymentStatus().success( null, {
+				setPaymentStatus().success( {
```

Comment:
From my fiddling yesterday I think we might need to also `setActivePaymentMethod()` here. That should fix the issue with selecting `Cheque payments` tab then selecting a saved stripe card.

Reply:
Also wondered about switching this argument list to an object, named-parameters style, to protect against "argument in wrong position" regressions. Not sure if this is risky / better but maybe worth considering.

e.g. something like this

```js
setPaymentStatus().success( {
  paymentMethodData: {
     payment_method: activePaymentMethod,
     savedPaymentMethodToken: token,
  }
} );
```

Reply:
> From my fiddling yesterday I think we might need to also setActivePaymentMethod() here.

The reason this worked is due to happy circumstance with an existing bug (`activePaymentMethod` should be a dependency). It's actually the incorrect fix here because activePaymentMethod will be whatever is selected in the tab, instead we need to update the activePaymentMethod via the selected payment option. I've got a fix underway for that.

> Also wondered about switching this argument list to an object, named-parameters style, to protect against "argument in wrong position" regressions. Not sure if this is risky / better but maybe worth considering.

What places in the code are dependent on the order of the properties? The argument **is** [`paymentMethodData`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/81b4633ae78d8903cb866f9579e95590337bbcab/assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js#L252-L256).

Reply:
> What places in the code are dependent on the order of the properties? The argument is paymentMethodData.

The original issue looked like an arg-order issue to me - we were passing `paymentMethodData` as arg 2, but it was arg 1 (fix was to remove the `null` placeholder for arg 1). If we made the func take an object, i.e. named parameters, we could reduce risk of that kind of bug in future. No biggie, not a blocker.

Reply:
Oooohhh, sorry I misunderstood, I thought you were referring to something with the _property_ order (and missed the important usage of the word _argument_ in your comment). Hmm, ya I can see your point, but beyond that, there isn't much compelling reason to me for the resulting code churn at the moment to do that.